lsp-common.el

```diff
@@ -40,14 +40,15 @@
 
 (defvar lsp--no-response)
 
-;; from http://emacs.stackexchange.com/questions/8082/how-to-get-buffer-position-given-line-number-and-column-number
 (defun lsp--position-to-point (params)
   "Convert Position object in PARAMS to a point."
   (save-excursion
-    (goto-char (point-min))
-    (forward-line (gethash "line" params))
-    (forward-char (gethash "character" params))
-    (point)))
+    (save-restriction
+      (widen)
+      (goto-char (point-min))
+      (goto-char (+ (line-beginning-position (1+ (gethash "line" params)))
+                    (gethash "character" params)))
+      (point))))
```

Comment:
Please add a comment why we use `goto-char` and `point` instead of just returning the value.